impeller/display_list/nine_patch_converter.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <vector>
+
+#include "impeller/display_list/nine_patch_converter.h"
+
+namespace impeller {
+
+NinePatchConverter::NinePatchConverter() = default;
+
+NinePatchConverter::~NinePatchConverter() = default;
+
+std::vector<double> NinePatchConverter::InitSlices(double img0,
+                                                   double imgC0,
+                                                   double imgC1,
+                                                   double img1,
+                                                   double dst0,
+                                                   double dst1) {
+  auto imageDim = img1 - img0;
+  auto destDim = dst1 - dst0;
+
+  if (imageDim == destDim) {
+    // If the src and dest are the same size then we do not need scaling
+    // We return 4 values for a single slice
+    return {img0, dst0, img1, dst1};
+  }
+
+  auto edge0Dim = imgC0 - img0;
+  auto edge1Dim = img1 - imgC1;
+  auto edgesDim = edge0Dim + edge1Dim;
+
+  if (edgesDim >= destDim) {
+    // the center portion has disappeared, leaving only the edges to scale to a
+    // common center position in the destination this produces only 2 slices
+    // which is 8 values
+    auto dstC = dst0 + destDim * edge0Dim / edgesDim;
+    return {
+        img0, dst0, imgC0, dstC, imgC1, dstC, img1, dst1,
```

Comment:
If these return vectors could be initialized 4 values at a time it is easier to read the slices out and see how they relate.

Reply:
Done